src/server/room-controller.js

```diff
@@ -86,6 +73,34 @@ export default class RoomController {
     return res.status(201).end();
   }
 
+  async handleGetRoomPage(req, res) {
+    const { roomId } = req.params;
+    const room = await this.roomService.getRoomById(roomId);
+
+    if (!room) {
+      throw new NotFound();
+    }
+
+    const { _id: userId } = req.user;
+
+    let invitations = null;
```

Comment:
Another small thing:
For me this would make more sense to be an empty array by default. Then the page can make `invitations` required and knows it always has to deal with an array and can check for empty array when it needs a special case.

Reply:
The invitations are not required by the room page. They are only loaded when you are the room owner, thus preventing members from ever seeing invitations. Here are the propTypes:
 invitations: PropTypes.arrayOf(invitationShape) //no isRequired added
I've put  room: roomShape.isRequired, that was forgotten and I think this caused the confusion, thanks!

Reply:
Yes I know that they are, as data, currently not required, there was no confusion 👍 

What I said was that the page could simply always require an `invitations` array there and display what it gets (an empty array being, still, a valid array and fulfilling the requirement).
Because the room page itself does not care about the logic of room owner or not when it comes to invitations, it just displays. The business logic of populating `invitations` or not is strictly in the controller.

Reply:
The UI is a presentation layer and it should present the data sent by the controller to the user. 
The controller decides if you deliver invitations or not based on the type of user (member or owner).
The UI just checks if there are invitations delivered or not and displays them if they are (it doesn't care about owner or member as you said).

If we want to deliver an empty array we need to deliver an extra bit of information (or make the UI smarter) to know whether it should display the invitations section or not. Maybe something like:
{ 
  isAllowedToViewInvitations: true/false,
  invitations: [...]
}

Would this be better?

Reply:
I think this complicates things. Why not simply have the controller always send an array and have the UI always display the collapsible invitations section, which in some cases will simply be empty. But this way the UI is also consistent and the logic is as simple as can be.

Reply:
Currently the invitations section has also the send invite button, we'd need to hide that for members. Plus it displays an empty section that will always be empty for a normal member.

For me right now the UI is the simplest it can be: if there is data display it, if not, then don't.

Reply:
Ok then let's just leave this part as is now and not drag this thread longer 😬

Reply:
We'll have to change this pattern anyway (probably), as there will be much more data in the room admin area (that is only visible to owners).

Reply:
I've changed it so that the room page decides what to show. I realized I needed also to not display this section for the public rooms (invitations make no sense there since everyone is a "member").

Sorry for the bid discussion, my head was not in the right place this morning.

Reply:
Hmm, actually I will hide the members list as well, it makes no sense on the public rooms. We can revisit this later.